Runtime/World/World.cpp

```diff
@@ -54,6 +48,19 @@ namespace Spartan
 		SUBSCRIBE_TO_EVENT(Event_World_Resolve, [this](Variant) { m_isDirty = true; });
 		SUBSCRIBE_TO_EVENT(Event_World_Stop,	[this](Variant)	{ m_state = Idle; });
 		SUBSCRIBE_TO_EVENT(Event_World_Start,	[this](Variant)	{ m_state = Ticking; });
+
+        // Hardcoded types make it harder to write new components
+        m_components_managers[ComponentType_Transform]      = std::make_shared<ComponentManager<Transform>>();
```

Comment:
There is a way to avoid these hardcoded types, because in some function, you know both the enumeration and the type, would you like to give it a try ?

Reply:
Yes I think that is possible. First i would create a global tuple with all the component types, and what I could do is to create a function like the IterateManagers one that would instead fill the map with the managers. But still this requires that tuple with hardcoded types. Also see my post below for furher details.